src/Components/AllListings.jsx

```diff
@@ -6,18 +6,31 @@ class AllListings extends React.Component {
   constructor(props) {
     super(props);
     this.scroller = React.createRef();
+    this.state = {
+      scrollPosition: 0,
+      maxScroll: Infinity,
+    }
   }
 
   render() {
     const { data } = this.props;
-    if (data) {
+    const { scrollPosition } = this.state;
+    const { maxScroll } = this.state;
+
+    if (data) {   
       return (
         <div className="allListings" ref="scroller">
-          <button type="button" id="leftScroll" onClick={(e)=>{ this.refs.scroller.scrollLeft -= 400; }}>{'<'}</button>
+          {
+            scrollPosition > 25 && 
+            <button type="button" id="leftScroll" onClick={(e)=>{this.setState({scrollPosition: scrollPosition - 400, maxScroll: this.refs.scroller.scrollWidth - this.refs.scroller.clientWidth }, () => {this.refs.scroller.scrollLeft -= 400 }); }}>{'<'}</button>

```

Comment:
This line is very long, consider entering each option on a new line. It would have the same tab spacing as the tag it is in.